org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeBinding.java

```diff
@@ -399,8 +399,7 @@ public boolean isFromSource() {
 
 	@Override
 	public boolean isGenericType() {
-		// TODO Auto-generated method stub
-		throw new UnsupportedOperationException("Unimplemented method 'isGenericType'");
+		return this.typeSymbol.asType().isParameterized();
```

Comment:
are you sure about this one? generic != parameterized. `Foo<T>` is generic, `Foo<String>` is parameterized

Reply:
You're right, this is wrong. I'll need to look into this more, since there is no quick fix. I don't think that javac distinguishes between type parameters and type arguments, or maybe it just doesn't store type parameters.

Reply:
I made the same mistake before ;-)

Reply:
semantic highlighting breaks in jdt.ls because of this being non-implemented yet

```
Caused by: java.lang.UnsupportedOperationException: Unimplemented method 'isGenericType'
	at org.eclipse.jdt.internal.javac.dom.JavacTypeBinding.isGenericType(JavacTypeBinding.java:403)
	at org.eclipse.jdt.ls.core.internal.semantictokens.TokenModifier.checkGeneric(TokenModifier.java:149)
	at org.eclipse.jdt.ls.core.internal.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:320)
	at org.eclipse.jdt.core.dom.SimpleName.accept0(SimpleName.java:198)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChild(ASTNode.java:3382)
	at org.eclipse.jdt.core.dom.SimpleType.accept0(SimpleType.java:195)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChild(ASTNode.java:3382)
	at org.eclipse.jdt.core.dom.ArrayType.accept0(ArrayType.java:247)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChild(ASTNode.java:3382)
	at org.eclipse.jdt.core.dom.SingleVariableDeclaration.accept0(SingleVariableDeclaration.java:394)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:3405)
	at org.eclipse.jdt.core.dom.MethodDeclaration.accept0(MethodDeclaration.java:670)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.ls.core.internal.semantictokens.SemanticTokensVisitor.acceptNodeList(SemanticTokensVisitor.java:669)
	at org.eclipse.jdt.ls.core.internal.semantictokens.SemanticTokensVisitor.visit(SemanticTokensVisitor.java:493)
	at org.eclipse.jdt.core.dom.TypeDeclaration.accept0(TypeDeclaration.java:481)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.core.dom.ASTNode.acceptChildren(ASTNode.java:3405)
	at org.eclipse.jdt.core.dom.CompilationUnit.accept0(CompilationUnit.java:258)
	at org.eclipse.jdt.core.dom.ASTNode.accept(ASTNode.java:3334)
	at org.eclipse.jdt.ls.core.internal.handlers.SemanticTokensHandler.full(SemanticTokensHandler.java:57)
```

You can probably implement something similar to https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/blob/c769f468ee881ff300bd3fc82a446071c60821dd/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/TypeBinding.java#L1015-L1022